parquet-column/src/main/java/org/apache/parquet/internal/column/columnindex/ColumnIndexBuilder.java

```diff
@@ -543,6 +546,11 @@ public static ColumnIndex build(
    *          the statistics to be added
    */
   public void add(Statistics<?> stats) {
+    if (stats.isEmpty()) {
```

Comment:
This is used by the rewriter to rebuild the ColumnIndex. The change enables it to invalidate the ColumnIndex if any statistics is invalid, right? This is also used by the general file writer. Does it affect the behavior? If yes, probably we need add a test case.

Reply:
Reverted these changes